tests/bsim/bluetooth/mesh/src/friendship_common.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "friendship_common.h"
+#include "mesh_test.h"
+
+#define LOG_MODULE_NAME friendship_common
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+uint16_t bt_mesh_test_friend_lpn_addr;
+ATOMIC_DEFINE(bt_mesh_test_state, BT_MESH_TEST_FRIENDSHIP_FLAGS);
+struct k_sem bt_mesh_test_friendship_events[BT_MESH_TEST_FRIENDSHIP_FLAGS];
+
+void evt_signal(enum bt_mesh_test_friendship_evt_flags evt)
+{
+	atomic_set_bit(bt_mesh_test_state, evt);
+	k_sem_give(&bt_mesh_test_friendship_events[evt]);
```

Comment:
It seems a bit strange. You have semaphore and states(aka bitmap) that are not encapsulated within this file but you created a set of functions to manipulate them.

If you want to create a separate module with common friendship functionality, please make it with the correct splitting. There should be static parameters that have access via module API. Now it seems like splitting the file into two pieces (that is the truth). All variables are available everywhere. It will be very difficult to follow up usage of this and its maintenance.

This module requires more work.

Reply:
I made `bt_mesh_test_friendship_state` and `bt_mesh_test_friendship_events` static, now only exposed API functions have access to them. Added function to initialize semaphores and check state. Only `bt_mesh_test_friend_lpn_addr` variable is accessible from API, but maybe it's worth hiding it also - I can create getter function for it

Reply:
It will be more correct to have a getter instead of extern global variable.